lib/puma/thread_pool.rb

```diff
@@ -424,6 +429,9 @@ def shutdown(timeout=-1)
 
         # Wait +timeout+ seconds for threads to finish.
         join.call(timeout)
+        if @shutdown_debug == :on_force && !threads.empty?
+          shutdown_debug("Shutdown timeout exceeded")
```

Comment:
Unlike the graceful shutdown case where we print before we join on the threads (which makes sense because the join will wait for as long as needed), here we print after joining - once if and before forcing a shutdown, and then again if and before forcefully killing threads.